apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/faucet/components/SpinWheel.tsx

```diff
@@ -0,0 +1,191 @@
+
+const segments = [
+  { segmentText: "Come back tomorrow", segColor: "#f771c9" }, // Mid-dark, optimized for dark mode
+  { segmentText: "0.1", segColor: "#f65ac0" }, // Lightest, optimized for dark mode
+  { segmentText: "Come back tomorrow", segColor: "#f542b7" }, // Slightly darker, optimized
+  { segmentText: "0.01", segColor: "#f32bae" }, // Mid-light, optimized for dark mode
+  { segmentText: "0.05", segColor: "#f213a5" }, // Mid, optimized for dark mode
+  { segmentText: "0.01", segColor: "#da1195" }, // Mid-dark, optimized for dark mode
+  { segmentText: "0.05", segColor: "#c20f84" }, // Darker, optimized for dark mode
+  { segmentText: "0.01", segColor: "#a90d73" }, // Even darker, optimized
+  { segmentText: "0.05", segColor: "#910b63" }, // Near darkest, optimized for dark mode
+  { segmentText: "0.01", segColor: "#790a53" }, // Lightest, optimized for dark mode
+  { segmentText: "0.05", segColor: "#610842" }, // Slightly darker, optimized
+  { segmentText: "0.01", segColor: "#490632" }, // Mid-light, optimized for dark mode
+];
+
+function formatTime(seconds: number) {
+  const hours = Math.floor(seconds / 3600);
+  const minutes = Math.floor((seconds % 3600) / 60);
+
+  if (hours === 0 && minutes <= 1) {
+    return "1 minute";
+  }
+
+  if (seconds > 60 && seconds < 3600) {
+    return `${minutes}m`;
+  }
+
+  return `${hours}h ${minutes}m`;
+}
+
+export function TestnetSpinWheel({
+  chain,
+  ttlSeconds,
+}: {
+  chain: ChainMetadataWithServices;
+  ttlSeconds: number;
+}) {
+  const address = useActiveAccount()?.address;
+  const chainId = chain.chainId;
+  const definedChain = defineChain(chainId);
+  const { data: evmBalance, isLoading } = useWalletBalance({
+    address: THIRDWEB_ENGINE_FAUCET_WALLET,
+    chain: definedChain,
+    client: thirdwebClient,
+  });
+  const router = useRouter();
+
+  const claimMutation = useMutation({
+    mutationFn: async (result: string) => {
+      try {
+        const amount = result === "Come back tomorrow" ? "0" : result;
+        const requestBody = {
+          chainId: chainId,
+          toAddress: address,
+          amount: amount,
+        };
+
+        const response = await fetch("/api/testnet-faucet/claim", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(requestBody),
+        });
+
+        if (!response.ok) {
+          const errorText = await response.json();
+          toast.error(errorText.error);
+          return;
+        }
+
+        router.refresh();
+        const successMessage = `${amount} ${chain?.nativeCurrency?.name} sent to ${address}`;
+        toast.success(successMessage);
+      } catch (error) {
+        toast.error("An unexpected error occurred.");
+        throw error;
+      }
+    },
+  });
+
+  const isFaucetEmpty =
+    evmBalance?.value !== undefined && evmBalance.value < 100000000000000000n;
```

Comment:
this basically is checking if it's below `1 "ETH"`? if so you should be able to use the `toUnits("1", 18)` from `thirdweb/utils` for this

Reply:
done